### ISSUE TITLE
Using a term model in setModelOptions will return all terms for all taxonomies

### ISSUE THREAD

**Issue by bucur-marian** [TypeRocket/typerocket#182]
Doing something like:

```php
echo $form->select('Foo')->setModelOptions( new Foo(), 'name', 'term_id' );
```
will render a select populate with all terms from all taxonomies, not just the `Foo` taxonomy.

Tested with Typerocket `v4.0.0-beta26`.

**Comment by bucur-marian** [comment]
Actually, the issue is with `findAll` and `get` when called for a term model.

E.g.:  `( new Foo() )->findAll()->get()` returns all terms, not just `Foo`'s terms.

**Comment by kevindees** [comment]
Hey @mbucurcf 

I'm working on a more advanced term model to help address issues like this. If you like you can try this branch https://github.com/TypeRocket/core/tree/advanced-term-models for core while I work this system out. I'm not sure if it breaks anything yet.

**Comment by kevindees** [comment]
@mbucurcf 

This is now a patch fix in v4.0.4 of core. Let me know if that worked for you.

https://github.com/TypeRocket/core/releases/tag/v4.0.4
